Unified Vulkan build fixes

nsubtil opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/Vulkan-ValidationLayers?pullRequest=455) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/Vulkan-ValidationLayers?pullRequest=455) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/Vulkan-ValidationLayers?pullRequest=455) it.</sub>

mikew-lunarg commented on tests/CMakeLists.txt:
```diff
@@ -72,15 +72,15 @@ set(COMMON_CPP vkrenderframework.cpp vktestbinding.cpp vktestframework.cpp test_
     # extra setup for out-of-tree builds
     if(NOT (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_CURRENT_BINARY_DIR))
-        add_custom_target(binary-dir-symlinks ALL COMMAND ln -sf ${CMAKE_CURRENT_SOURCE_DIR}/run_all_tests.sh VERBATIM)
+        add_custom_target(VulkanVL_binary-dir-symlinks ALL COMMAND ln -sf ${CMAKE_CURRENT_SOURCE_DIR}/run_all_tests.sh VERBATIM)
```
I like adding the prefix.  Could you please convert the hyphens to underscores for consistency?

mikew-lunarg commented on CMakeLists.txt:
```diff
@@ -137,120 +145,136 @@ option(BUILD_LAYERS "Build layers" ON)
 if(NOT GLSLANG_INSTALL_DIR)
     set(GLSLANG_INSTALL_DIR $ENV{GLSLANG_INSTALL_DIR})
 endif()
-message(STATUS "Using glslang install located at ${GLSLANG_INSTALL_DIR}")
```
This large diff block appears caused by whitespace due to wrapping with an if().  It would be helpful for reviewing to have the whitespace changes as a separate commit.

mikew-lunarg replied:
Yes, separate and smaller commits make reviews much easier (I've had several dozen in a PR).
After the PR passes review, then squash them all down to a single commit, which concatenates the comments for a detailed record.

nsubtil replied:
Is this change really that big, or is the concern just the whitespace block?

This seems roughly in-line with what I'm used to in other projects, which is why I'm asking. I'm trying to tease apart the whitespace changes in a meaningful way now.

mikew-lunarg replied:
I'm not suggesting you separate this PR, if my assumption in this case is correct that the big block is just whitespace, without little changes hiding within.

nsubtil replied:
I've verified locally via a diff viewer with an option to hide whitespace. There are no changes in this block beyond indentation/whitespace + moving variables into different `if (NOT TARGET ...)` blocks to account for the case where a given dependency was already imported into the build.

Will keep your suggestion in mind for future reviews.

mikew-lunarg commented:
Please format per the "Commit Messages" section of [CONTRIBUTING.md](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/CONTRIBUTING.md)

nsubtil commented:
This is now rebased on master, and I believe I should have addressed the review feedback:

- Commit messages (hopefully) fixed to comply with guidelines
- Renamed the intermediate target to use all underscores

@mikew-lunarg would you mind taking a look to see if anything else is required?

Unfortunately I forgot to switch the base branch on GitHub prior to pushing out the rebase, which means AppVeyor decided the PR won't merge cleanly and can't be built. Is there a way to manually trigger the automation here?

mikew-lunarg commented:
@nsubtil Before pushing, please squash the two commits into one and deWIP.  That, and rebasing on the new master, should relaunch AppVeyor?

I've done that to run on our internal CI, and will report the results.

mikew-lunarg commented:
@mark-lunarg FYI

mikew-lunarg commented:
Looks fine.  Cheers

mikew-lunarg commented:
Pushed to master
